src/clandinin_lab_to_nwb/brezovec/brezovecimaginginterface.py

```diff
@@ -1,21 +1,119 @@
 from dateutil.parser import parse
 from clandinin_lab_to_nwb.brezovec.brezovecimagingextractor import BrezovecMultiPlaneImagingExtractor
 
-from roiextractors import MultiImagingExtractor
 from neuroconv.datainterfaces.ophys.baseimagingextractorinterface import BaseImagingExtractorInterface
 from neuroconv.utils import FolderPathType
 from neuroconv.utils.dict import DeepDict
 
 
-class BrezovecMultiPlaneImagingInterface(BaseImagingExtractorInterface):
+class BrezovecFunctionalGreenImagingInterface(BaseImagingExtractorInterface):
     """
-    Data Interface for writing the Clandinin lab imaging data to NWB file using BrezovecMultiPlaneImagingExtractor.
+    Data Interface for writing the functional imaging data (GCaMP6f) for Clandinin lab to NWB file using BrezovecMultiPlaneImagingExtractor.
     """
 
     Extractor = BrezovecMultiPlaneImagingExtractor
 
     def __init__(self, folder_path: FolderPathType, stream_name: str, verbose: bool = True):
+        """
+        Initialize reading of NIfTI files.
 
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            The path to the folder that contains the NIfTI image files (.nii) and configuration files (.xml) from Bruker system.
+        stream_name: str, optional
+            The name of the recording channel.
+        verbose : bool, default: True
+        """
+        super().__init__(
+            folder_path=folder_path,
+            stream_name=stream_name,
+            verbose=verbose,
+        )
+
+    @classmethod
+    def get_streams(cls, folder_path) -> dict:
+        streams = BrezovecMultiPlaneImagingExtractor.get_streams(folder_path=folder_path)
+        return streams
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+
+        xml_metadata = self.imaging_extractor.xml_metadata
+        session_start_time = parse(xml_metadata["date"])
+        metadata["NWBFile"].update(session_start_time=session_start_time)
+
+        # channel_name = 'Green'
+        # metadata["Ophys"]["OpticalChannel"][0].update(
+        #     name=channel_name,
+        #     emission_lambda=513.0,
+        #     description="Green channel of the microscope, 525/50 nm filter.",
+        # )
+
+        device_name = "BrukerFluorescenceMicroscope"
+        description = f"Version {xml_metadata['version']}"
+        metadata["Ophys"]["Device"][0].update(
+            name=device_name,
+            description=description,
+        )
+
+        imaging_plane_name = "ImagingPlaneGCaMP6fFunctional"
+        imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][
+            0
+        ]  # should it be photon_series_index pr imaging_plane_index
+        imaging_plane_metadata.update(
+            name=imaging_plane_name,
+            # optical_channel=channel_name,
+            device=device_name,
+            excitation_lambda=920.0,
+            indicator="GCaMP6f",
+            imaging_rate=self.imaging_extractor.get_sampling_frequency(),
```

Comment:
We don't need this. The pynwb documentation says that if it is on the PhotonSeries it can be ommited here.

Reply:
https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html#pynwb.ophys.ImagingPlane